doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+builtin basic function and let the user compose them to meet its need.</why>
+<ko>The reference application supports in a tricky way: the logarithm
+adaptation is always combined with the affine adaptation</ko>
+
+## Identifiers
+Every parameter **MUST** have an identifier that uniquely identifies it.
+<why>to identify a parameter outside the framework</why>
+
+This identifier **SHOULD** be a string.
+<why>so that the (human</why> user can identify a parameter with ease.)
+
+Two PF instances with the same parameters **MUST** have the same identifier for
+those parameters.
+I.e. this identifier should be the same across all instances with the same
+configuration.
+<why>Persistence of parameter identifier across PF instances with the same
+configuration. To identify parameters independently of the host machine and PF
+instance</why>
+
+### Parameter tree
+A parameter **MUST** be structured in a tree. Each parameter being a distinct
+tree leaf.
+<why>Tree is a simple data structure that can be easily represented and is
+enough to map underlined layers.</why>
+
+Each node of the tree **SHOULD** have its own identifier with the same
+characteristics (type, independence...) than a parameter.
+<why>To represent the tree without treating the leaf nodes specifically.</why>
+
+The identifier of each node of the tree **SHOULD** be a combination of its
+parents. More specifically, if the identifier is a string it should be
+formated the same way as a file system path. E.g. the parameter named
+`/root/child1/4/parameter1`.
+<why>Usual syntax to address trees.</why>
+
+
+# Syncer
+
+The PF philosophy is to map the hardware characteristics to parameters.
+
+A syncer **MUST** be mapped to one or more parameters.
+<why>The hardware minimal access may be bigger than one parameter.</why>
+
+One parameter **MUST NOT** be mapped to two or more syncer.
+<why>Which syncer should be responsible to retrieve the initial parameter value
+if they are multiple per parameter?</why>
+
+A syncer **MUST** support retrieving the mapped parameters value from the mapped
+hardware.
+<why>to retrieve a parameter value at the start of the PF.</why>
+
+A syncer **MUST** support setting the mapped parameters value to the mapped
+hardware.
+<why>to synchronise hardware on parameter change.</why>
+
+Syncers **MUST** retrieve and set the parameters value from the PF core.
+<why>to be able to synchronise with hardware.</why>
+
+This API **MAY** be a packed parameter structure, following the C ABI without
+padding.
+<nb>This is what the reference implementation does.</nb>
+<why>???</why>
+
+## Introspection
+The syncer API **SHOULD** allow introspection of the mapped parameters.
+<why>the parameter structure may be useful for the syncer to communicate with
+the hardware.</why>
+
+## Plugins
+The PF **MUST** be able to create syncers.
+<why>to bind on the corresponding parameters.</why>
+
+### Definition
+The PF creates syncer using syncer builder.
```

Comment:
It is a definition (as explained in the title of the section) that introduces the concept of syncer builder.